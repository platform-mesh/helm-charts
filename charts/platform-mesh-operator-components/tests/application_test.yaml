# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test the application manifests
release:
  name: operator-components
  namespace: openmfp-system
tests:
- it: it should render the application manifests
  set:
    componentVersion:
      semver: 0.0.0
  asserts:
  - matchSnapshot: {}
