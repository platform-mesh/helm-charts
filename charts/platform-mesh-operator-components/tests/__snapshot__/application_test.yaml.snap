it should render the application manifests:
  1: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: image
        component: etcd-druid
        repo: oci
      name: etcd-druid-image
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: etcd-druid
          resource:
            name: image
  2: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: account-operator
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: account-operator
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: account-operator
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        crds:
          enabled: false
        kcp:
          apiExportEndpointSliceName: ""
          enabled: true
        kubeconfigSecret: account-operator-kubeconfig
        log:
          level: debug
        subroutines:
          fga:
            grpcAddr: openfga:8081
        tracing:
          collector:
            host: observability-opentelemetry-collector.observability.svc.cluster.local:4317
          enabled: true
  3: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: crossplane
      namespace: default
    spec:
      chart:
        spec:
          chart: crossplane
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: HelmRepository
            name: crossplane
      dependsOn: null
      interval: 1m
      releaseName: crossplane
      targetNamespace: crossplane-system
      timeout: 5m
      values:
        provider:
          packages:
            - xpkg.upbound.io/crossplane-contrib/provider-keycloak:v1.8.0
  4: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: etcd-druid
      namespace: default
    spec:
      chart:
        spec:
          chart: charts
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: GitRepository
            name: etcd-druid
      dependsOn: null
      interval: 1m
      releaseName: etcd-druid
      targetNamespace: etcd-druid-system
      timeout: 5m
      values: {}
  5: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: extension-manager-operator
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: extension-manager-operator
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: extension-manager-operator
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        crds:
          enabled: false
        kcp:
          enabled: true
          kubeconfigSecret: extension-manager-operator-kubeconfig
        tracing:
          collector:
            host: observability-opentelemetry-collector.observability.svc.cluster.local:4317
          enabled: true
  6: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: iam-service
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: iam-service
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
        - name: openfga
          namespace: default
      interval: 1m
      releaseName: iam-service
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        gateway:
          name: gateway
        hostname: api.example.com
        trust:
          default:
            audience: default
            jwksUrl: http://keycloak-headless.platform-mesh-system:8080/keycloak/realms/default/protocol/openid-connect/certs
            trustedIssuer: https://example.com:443/keycloak/realms/default
  7: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: iam-ui
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: iam-ui
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: iam-ui
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        istio:
          virtualService:
            hosts:
              - example.com
  8: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: infra
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: infra
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
        - name: kcp-operator
          namespace: default
      interval: 1m
      releaseName: infra
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        istio:
          main:
            gateway:
              hosts:
                - example.com
                - '*.example.com'
              name: https
              port: "443"
              protocol: HTTPS
              tls:
                credentialName: domain-certificate
                minProtocolVersion: TLSV1_2
                mode: SIMPLE
          passThrough:
            gateway:
              enabled: true
              hosts:
                - kcp.api.example.com
              name: pass-https
              port: "443"
              protocol: HTTPS
        keycloak:
          istio:
            virtualservice:
              hosts:
                - example.com
  9: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: istio-base
      namespace: default
    spec:
      chart:
        spec:
          chart: base
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: HelmRepository
            name: istio-base
      dependsOn: null
      interval: 1m
      releaseName: istio-base
      targetNamespace: istio-system
      timeout: 5m
      values: null
  10: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: istio-gateway
      namespace: default
    spec:
      chart:
        spec:
          chart: gateway
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: HelmRepository
            name: istio-gateway
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: istio-gateway
      targetNamespace: istio-system
      timeout: 5m
      values:
        service:
          ports:
            - name: https
              nodePort: 31000
              port: 8443
            - name: status-port
              nodePort: 32000
              port: 15021
          type: NodePort
  11: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: istio-istiod
      namespace: default
    spec:
      chart:
        spec:
          chart: istiod
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: HelmRepository
            name: istio-istiod
      dependsOn:
        - name: istio-base
          namespace: default
      interval: 1m
      releaseName: istio-istiod
      targetNamespace: istio-system
      timeout: 5m
      values:
        meshConfig:
          defaultConfig:
            holdApplicationUntilProxyStarts: true
            tracing:
              provider:
                name: otel-tracing
          extensionProviders:
            - name: otel-tracing
              opentelemetry:
                port: 4317
                protocol: grpc
                service: observability-opentelemetry-collector.observability.svc.cluster.local
        tracing:
          enabled: false
          telemetry:
            tracing:
              - providers:
                  - name: otel-tracing
                randomSamplingPercentage: 100
  12: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: kcp-operator
      namespace: default
    spec:
      chart:
        spec:
          chart: kcp-operator
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: HelmRepository
            name: kcp-operator
      dependsOn: null
      interval: 1m
      releaseName: kcp-operator
      targetNamespace: kcp-operator
      timeout: 5m
      values: null
  13: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: keycloak
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: keycloak
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: keycloak
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        auth:
          adminUser: keycloak-admin
          existingSecret: keycloak-admin
          passwordSecretKey: secret
        extraEnvVars:
          - name: KEYCLOAK_USER
            value: keycloak-admin
          - name: KEYCLOAK_PASSWORD
            valueFrom:
              secretKeyRef:
                key: secret
                name: keycloak-admin
          - name: JAVA_OPTS_APPEND
            value: -Djgroups.dns.query=openmfp-keycloak-headless.openmfp-system.svc.cluster.local
        httpRelativePath: /keycloak/
        postgresql:
          auth:
            existingSecret: ""
            secretKeys:
              adminPasswordKey: password
              userPasswordKey: password
            username: keycloak
          nameOverride: postgresql-keycloak
          primary:
            resourcesPreset: none
  14: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: kubernetes-graphql-gateway
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: kubernetes-graphql-gateway
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: kubernetes-graphql-gateway
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        kubeConfig:
          enabled: true
          secretName: kubernetes-grapqhl-gateway-kubeconfig
        listener:
          virtualWorkspacesConfig:
            content:
              virtualWorkspaces:
                - kubeconfig: /app/kubeconfig/kubeconfig
                  name: contentconfigurations
                  url: https://kcp-front-proxy.platform-mesh-system:8443/services/contentconfigurations
                - kubeconfig: /app/kubeconfig/kubeconfig
                  name: marketplace
                  url: https://kcp-front-proxy.platform-mesh-system:8443/services/marketplace
            enabled: true
        trust:
          default:
            audience: default
            jwksUrl: http://keycloak-headless.platform-mesh-system:8080/keycloak/realms/default/protocol/openid-connect/certs
            trustedIssuer: https://example.com:443/keycloak/realms/default
        virtualService:
          hosts:
            - example.com
            - '*.example.com'
          httpRules:
            - cors:
                allowHeaders:
                  - '*'
                allowMethods:
                  - GET
                  - POST
                allowOrigins:
                  - regex: .*
              name: default
          pathPrefix: /api/kubernetes-graphql-gateway/
  15: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: openfga
      namespace: default
    spec:
      chart:
        spec:
          chart: openfga
          reconcileStrategy: ChartVersion
          sourceRef:
            kind: HelmRepository
            name: openfga
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: openfga
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        autoscaling:
          enabled: false
        checkQueryCache:
          enabled: true
          limit: 10000
          ttl: 10s
        datastore:
          applyMigrations: true
          engine: postgres
          maxOpenConns: 30
          migrationType: initContainer
          migrations:
            image:
              pullPolicy: Always
              repository: groundnuty/k8s-wait-for
              tag: v2.0
        extraEnvVars:
          - name: OPENFGA_EXPERIMENTALS
            value: enable-list-users
        image:
          repository: openfga/openfga
          tag: ""
        log:
          level: info
        migrate:
          annotations:
            sidecar.istio.io/inject: "false"
        podAnnotations:
          traffic.sidecar.istio.io/excludeInboundPorts: "2112"
        postgresql:
          enabled: true
          nameOverride: postgres
        replicaCount: 1
        telemetry:
          trace:
            enabled: true
            otlp:
              endpoint: observability-opentelemetry-collector.openmfp-observability.svc.cluster.local:4317
              tls:
                enabled: false
  16: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: rebac-authz-webhook
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: rebac-authz-webhook
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: rebac-authz-webhook
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        certManager:
          createCA: true
          enabled: true
        istio:
          dnsNames:
            - rebac-authz-webhook.platform-mesh-system.svc.cluster.local
          exposed: false
        log:
          level: debug
        openfga:
          url: openfga:8081
  17: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: security-operator
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: security-operator
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: security-operator
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        crds:
          enabled: false
        fga:
          target: openfga.platform-mesh-system.svc.cluster.local:8081
        initializer:
          kubeconfigSecret: security-initializer-kubeconfig
        kubeconfigSecret: security-operator-kubeconfig
        log:
          level: debug
        operator:
          maxConcurrentReconciles: 1
          shutdownTimeout: 1m
  18: |
    apiVersion: helm.toolkit.fluxcd.io/v2
    kind: HelmRelease
    metadata:
      name: virtual-workspaces
      namespace: default
    spec:
      chartRef:
        kind: OCIRepository
        name: virtual-workspaces
        namespace: default
      dependsOn:
        - name: istio-istiod
          namespace: default
      interval: 1m
      releaseName: virtual-workspaces
      targetNamespace: platform-mesh-system
      timeout: 5m
      values:
        deployment:
          serverUrl: https://kcp-front-proxy.platform-mesh-system:8443
        service:
          port: 8443
  19: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Component
    metadata:
      name: platform-mesh
    spec:
      component: github.com/platform-mesh/platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      repositoryRef:
        name: platform-mesh
      semver: 0.0.0
  20: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Repository
    metadata:
      name: platform-mesh
    spec:
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      repositorySpec:
        baseUrl: ghcr.io/platform-mesh
        type: OCIRegistry
  21: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: account-operator
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: account-operator
          resource:
            name: chart
  22: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: helm
      name: crossplane
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: crossplane
          resource:
            name: chart
  23: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: git
      name: etcd-druid
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: etcd-druid
          resource:
            name: chart
  24: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: extension-manager-operator
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: extension-manager-operator
          resource:
            name: chart
  25: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: iam-service
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: iam-service
          resource:
            name: chart
  26: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: iam-ui
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: iam-ui
          resource:
            name: chart
  27: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: infra
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: infra
          resource:
            name: chart
  28: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: helm
      name: istio-base
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: istio-base
          resource:
            name: chart
  29: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: helm
      name: istio-gateway
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: istio-gateway
          resource:
            name: chart
  30: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: helm
      name: istio-istiod
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: istio-istiod
          resource:
            name: chart
  31: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: helm
      name: kcp-operator
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: kcp-operator
          resource:
            name: chart
  32: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: keycloak
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: keycloak
          resource:
            name: chart
  33: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: kubernetes-graphql-gateway
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: kubernetes-graphql-gateway
          resource:
            name: chart
  34: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: helm
      name: openfga
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: openfga
          resource:
            name: chart
  35: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: rebac-authz-webhook
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: rebac-authz-webhook
          resource:
            name: chart
  36: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: security-operator
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: security-operator
          resource:
            name: chart
  37: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: chart
        repo: oci
      name: virtual-workspaces
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - kind: Secret
          name: ocm-oci-github-pull
      resource:
        byReference:
          referencePath:
            - name: virtual-workspaces
          resource:
            name: chart
